src/main/webapp/map.js

```diff
@@ -87,9 +136,58 @@ function createMarker(place) {
     localStorage.setItem("store", place.place_id);
     shopInfo.open(map, marker);
   });
+  markers.push(marker);
 }
 
-function getShopName() {
-  document.getElementById('title').innerHTML = localStorage.getItem("shopName");
-  document.getElementById('address').innerHTML = localStorage.getItem("address");
-}
\ No newline at end of file
+function displayMarkers() {
+  for (var i = 0; i < markers.length; i++ ) {
+    markers[i].setMap(map);
+  }
+}
+
+function clearMarkers() {
+  for (var i = 0; i < markers.length; i++ ) {
+    markers[i].setMap(null);
+  }
+  markers = [];
+}
+
+/* Calculates the straight line distance between two sets of latitute and longitude */
+function haversine_distance(lat1,lon1,lat2,lon2) {
+  var R = 6371; // Radius of the earth in km
+  var dLat = deg2rad(lat2-lat1);  // deg2rad below
+  var dLon = deg2rad(lon2-lon1); 
+  var a = 
```

Comment:
what is `a`?

Reply:
It is a bit hard to give "a" a meaningful variable name since it's just a middle step in the formula. A breakdown of the formula can be found [here](http://www.movable-type.co.uk/scripts/latlong.html) The original math formula can be found [here](https://en.wikipedia.org/wiki/Haversine_formula)

Reply:
ok, can you add the first link in a comment above your code